### DISCUSSION TITLE
Why do users need to select a service for new claims?

### DISCUSSION THREAD

**Discussion by Sgitario** [halkyonio/primaza-poc#232] [Q&A]
From https://github.com/halkyonio/primaza-poc/issues/154, users need to select the service from a popup in the new claims form. 

Therefore, this new logic invalidates what the claiming service job is meant to do: search the services that have been requested in a claim using the "serviceRequested" property (something like "postgres-8"). This is because now we can directly link services to claim when creating a new one. 

Further clarification is required here to know how to approach this situation:
- Either allow both approaches: (1) select a service from a popup or (2) search the service using the "serviceRequested" property 
- Or remove the "serviceRequested" property and also the logic to search services
- Or revert the changes in https://github.com/halkyonio/primaza-poc/issues/154 and continue using the "serviceRequested" property

**Comment by aureamunoz** [comment]
I think the javadoc needs to be updated and aligned with the code in the [UpdateClaimJob](https://github.com/halkyonio/primaza-poc/blob/ebae32b55382372ef86af76e7c258987bf3a8404/servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java#L32).

That said, I will add comments to your propositions below: 
- _Either allow both approaches: (1) select a service from a popup or (2) search the service using the "serviceRequested" property_ --> currently this is what is implemented by code, no? we are doing (1) selecting a service from a list [gathered by services registered](https://github.com/halkyonio/primaza-poc/blob/88001e71e503d7ed2dbc830af0a998f0d05e7021/servicebox-app/src/main/resources/templates/claims/form.html#L35) in primaza **AND** we are doing (2) by [searching the service by `serviceRequested` ](https://github.com/halkyonio/primaza-poc/blob/ebae32b55382372ef86af76e7c258987bf3a8404/servicebox-app/src/main/java/io/halkyon/services/UpdateClaimJob.java#L58). Why do you propose to do (1) **OR** (2)?
- _Or remove the "serviceRequested" property and also the logic to search services_ --> in that case, who and how will set up the `claim.service` field? 

I wouldn't say that the new logic invalidates what the job does. The job#updateClaim does 2 things: link the claim with a service and increment attemps and update status. With the new logic, the job will still check if service is available and if a credential exists and modify the claim in consequence. I'm not sure to follow what you mean.

**Reply by Sgitario** [reply]
My point is that the linking of the claim-service is no longer needed, and also the `serviceRequested` property. Because we're already saying to what service the claim should be linked in the Claim form. 

The first approach is not implemented because in the Claim form we force users to select a service from the services popup. We could easily achieve this by either selecting one service or using the service requested expression (users would need to select one way or the another in the new claim form).

**Reply by aureamunoz** [reply]
OK so let's remove the `findService`action when the service is selected from a list of services and we can also support the scenario where the user can introduce manually a service name

**Reply by Sgitario** [reply]
PR: https://github.com/halkyonio/primaza-poc/pull/235
